contracts/DCAStrategies/DCAStrategies/DCAStrategiesManagementHandler.sol

```diff
@@ -127,10 +122,20 @@ abstract contract DCAStrategiesManagementHandler is IDCAStrategiesManagementHand
     if (_length > MAX_TOKEN_SHARES) revert TokenSharesExceedAmount(); // need to have less than the max
 
     uint256 _shares = 0;
-    for (uint256 i = 0; i < _length; i++) {
+    for (uint256 i = 0; i < _length; ) {
+      // isn't the last token
+      if (i < _length - 1) {
+        // ascending order ([1, 2, 3])
+        if (_tokens[i].token > _tokens[i + 1].token) revert TokenSharesNotSorted(); // not sorted
```

Comment:
So that we can detect and fail with duplicates also (please add a test for that too)
```suggestion
        if (_tokens[i].token >= _tokens[i + 1].token) revert TokenSharesNotSorted(); // not sorted
```